modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v550/RegistryDataManager.java

```diff
@@ -0,0 +1,112 @@
+import org.wso2.carbon.user.api.UserStoreException;
+
+import static org.wso2.carbon.base.MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+import static org.wso2.carbon.context.RegistryType.SYSTEM_GOVERNANCE;
+
+public class RegistryDataManager {
+
+    private static final Log log = LogFactory.getLog(RegistryDataManager.class);
+
+    public static final String ENTITLEMENT_POLICY_PUBLISHER = "/repository/identity/entitlement/publisher/";
+    public static final String PASSWORD_PROPERTY = "subscriberPassword";
+
+    private static RegistryDataManager instance = new RegistryDataManager();
+
+    private RegistryDataManager(){}
+
+    public static RegistryDataManager getInstance() {
+        return instance;
+    }
+
+    private void startTenantFlow(Tenant tenant) {
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        carbonContext.setTenantId(tenant.getId());
+        carbonContext.setTenantDomain(tenant.getDomain());
+    }
+
+    public void migrateSubscriberPassword(boolean migrateActiveTenantsOnly) throws UserStoreException {
+
+        //migrating super tenant configurations
+        try {
+            migrateSubscriberDataForTenant();
+            log.info("Policy Subscribers migrated for tenant : " + SUPER_TENANT_DOMAIN_NAME);
+        } catch (Exception e) {
+            log.error("Error while migrating Policy Subscribers for tenant : " + SUPER_TENANT_DOMAIN_NAME, e);
+        }
+
+        //migrating tenant configurations
+        Tenant[] tenants = ISMigrationServiceDataHolder.getRealmService().getTenantManager().getAllTenants();
+        for (Tenant tenant : tenants) {
+            if (migrateActiveTenantsOnly && !tenant.isActive()) {
+                log.info("Tenant " + tenant.getDomain() + " is inactive. Skipping Subscriber migration!!!!");
+                continue;
+            }
+            try {
+                startTenantFlow(tenant);
+                IdentityTenantUtil.getTenantRegistryLoader().loadTenantRegistry(tenant.getId());
+                migrateSubscriberDataForTenant();
+                log.info("Subscribers migrated for tenant : " + tenant.getDomain());
+            } catch (Exception e) {
+                log.error("Error while migrating Subscribers for tenant : " + tenant.getDomain(), e);
+            } finally {
+                PrivilegedCarbonContext.endTenantFlow();
+            }
+        }
+    }
+
+    private void migrateSubscriberDataForTenant() throws RegistryException, CryptoException {
+
+        Registry registry = PrivilegedCarbonContext.getThreadLocalCarbonContext().getRegistry(SYSTEM_GOVERNANCE);
+
+        if (registry.resourceExists(ENTITLEMENT_POLICY_PUBLISHER)) {
+            Collection subscriberCollection = (Collection) registry.get(ENTITLEMENT_POLICY_PUBLISHER);
+
+            for (String subscriberPath : subscriberCollection.getChildren()) {
```

Comment:
possible NPE

Reply:
In case of no child elements, an empty string array would be returned. Hence NPE is not possible